src/ui/foundation/GlobalCommands/GlobalCommands.tsx

```diff
@@ -1,13 +1,23 @@
-import React, { useContext } from 'react';
+import { useContext, useEffect } from 'react';
 
 import { SettingsContext } from '@/ui/foundation/SettingsProvider';
 import { useCommands } from '@/ui/hooks';
+import { GlobalStateContext } from '../GlobalStateProvider';
 
 /**
  * Register commands globally in the app.
  */
 export function GlobalCommands() {
+  const { commandEventTarget } = useContext(GlobalStateContext);
   const { settings } = useContext(SettingsContext);
+
   useCommands(settings.commands);
+
+  useEffect(() => {
+    settings.commands.forEach((command) =>
+      commandEventTarget.registerCommand(command)
+    );
+  }, [commandEventTarget, settings.commands]);
+
```

Comment:
This ensures that we don't call `registerCommand` again unless `settings.commands` changes.

Reply:
And what if it change? Should we clear the registered commands before registering again?

Reply:
It'll override the key with the updated value. We currently don't have a use case where we would want to unregister a command, but we'll append more commands, eg. within the editor.